src/s3/util/fileViewerManager.ts

```diff
@@ -65,13 +116,32 @@ export class S3FileViewerManager {
         }
         getLogger().verbose(`S3FileViewer: File from s3 or temp to be opened is: ${fileLocation}`)
 
-        const newTab = this.activeTabs.get(fileLocation.fsPath) ?? new S3Tab(fileLocation)
+        const newTab = this.activeTabs.get(fileLocation.fsPath) ?? new S3Tab(fileLocation, fileNode)
         await newTab.openFileInReadOnly()
 
         this.activeTabs.set(fileLocation.fsPath, newTab)
     }
 
     public async openInEditMode(uriOrNode: vscode.Uri | S3FileNode): Promise<void> {
+        if (this.promptOnEdit) {
+            const message = localize(
+                'AWS.s3.fileViewer.warning.editStateWarning',
+                'Opening S3 file for editing. Saved changes will be written directly to the original location in S3.'
+            )
+
+            const dontShow = localize('AWS.s3.fileViewer.button.dismiss', "Don't show this again")
+            const help = localize('AWS.s3.fileViewer.button.help', 'Help')
+
+            this.window.showWarningMessage(message, dontShow, help).then(selection => {
```

Comment:
Nit: show 'Close' option? I know you can just click the 'X' but it's nice to have a 'Close' button too.

Reply:
So have 3 buttons?

Reply:
I'd have to see the message for myself to know what makes sense for me. 3 buttons does sound like too much, and I don't think we can have a small help button ('?') either. I'm fine w/ the current implementation